src/app/http-interceptors/error.interceptor.spec.ts

```diff
@@ -0,0 +1,157 @@
+import { ErrorInterceptor } from './error.interceptor';
+import { TestBed } from '@angular/core/testing';
+import {
+    HttpClientTestingModule,
+    HttpTestingController,
+    TestRequest
+} from '@angular/common/http/testing';
+import {
+    HTTP_INTERCEPTORS,
+    HttpClient,
+    HttpErrorResponse
+} from '@angular/common/http';
+import { Data } from '@angular/router';
+import { AlertService } from '../components/alert/alert.service';
+import { MockTranslationModule } from '../testing/mock-translation-module';
+import { TranslationService } from 'angular-l10n';
+import Spy = jasmine.Spy;
+
+describe('ErrorInterceptor', () =>
+{
+    let httpClient:HttpClient;
+    let httpTestingController:HttpTestingController;
+    let alertService:AlertService;
+
+    beforeEach(() =>
+    {
+        TestBed.configureTestingModule({
+            imports:   [HttpClientTestingModule, MockTranslationModule],
+            providers: [{
+                provide:  HTTP_INTERCEPTORS,
+                useClass: ErrorInterceptor,
+                multi:    true
+            }, AlertService]
+        });
+
+        httpClient = TestBed.get(HttpClient);
+        httpTestingController = TestBed.get(HttpTestingController);
+        alertService = TestBed.get(AlertService);
+    });
+
+    it('should create an instance', () =>
+    {
+        let translationService:TranslationService = TestBed.get(TranslationService);
+        expect(new ErrorInterceptor(alertService, translationService)).toBeTruthy();
+    });
+
+    it(`should catch error status 401 - Unauthorized`, () =>
+    {
+        const errorMsg:string = 'Unauthorized';
+        const url:string = '';
+        let spy:Spy = spyOn(alertService, 'error');
+
+        httpClient.get<Data>(url).subscribe(
+            () => fail('should have failed with the 401 error'),
+            (error:HttpErrorResponse) =>
+            {
+                expect(error.status).toEqual(401);
+                expect(error.error).toEqual(errorMsg);
+                expect(spy).toHaveBeenCalled();
+            }
+        );
+
+        const request:TestRequest = httpTestingController.expectOne(url);
+        request.flush(errorMsg, {status: 401, statusText: 'Error'});
```

Comment:
This test should cover, that the event the redirect you to the login page should be fired.